From 0ae8e543792a2f7b635a941c10fc691e4316ac67 Mon Sep 17 00:00:00 2001
From: Andreas Monzner <andreas.monzner@dream-property.net>
Date: Mon, 20 Jan 2014 16:33:04 +0100
Subject: [PATCH] Revert "USB: ftdi_sio: fixed handling of unsupported CSIZE
 setting"

This reverts commit 75a4cfafc7379ed99dbb85f4b92de7e86a534876.

This change breaks smargo cardreader userspace software....
---
 drivers/usb/serial/ftdi_sio.c |   31 +++++++------------------------
 1 files changed, 7 insertions(+), 24 deletions(-)

diff --git a/drivers/usb/serial/ftdi_sio.c b/drivers/usb/serial/ftdi_sio.c
index 85504bb..a3f6fe0 100644
--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -2192,20 +2192,6 @@ static void ftdi_set_termios(struct tty_struct *tty,
 		termios->c_cflag |= CRTSCTS;
 	}
 
-	/*
-	 * All FTDI UART chips are limited to CS7/8. We won't pretend to
-	 * support CS5/6 and revert the CSIZE setting instead.
-	 */
-	if ((C_CSIZE(tty) != CS8) && (C_CSIZE(tty) != CS7)) {
-		dev_warn(&port->dev, "requested CSIZE setting not supported\n");
-
-		termios->c_cflag &= ~CSIZE;
-		if (old_termios)
-			termios->c_cflag |= old_termios->c_cflag & CSIZE;
-		else
-			termios->c_cflag |= CS8;
-	}
-
 	cflag = termios->c_cflag;
 
 	if (!old_termios)
@@ -2242,16 +2228,13 @@ no_skip:
 	} else {
 		urb_value |= FTDI_SIO_SET_DATA_PARITY_NONE;
 	}
-	switch (cflag & CSIZE) {
-	case CS7:
-		urb_value |= 7;
-		dev_dbg(&port->dev, "Setting CS7\n");
-		break;
-	default:
-	case CS8:
-		urb_value |= 8;
-		dev_dbg(&port->dev, "Setting CS8\n");
-		break;
+	if (cflag & CSIZE) {
+		switch (cflag & CSIZE) {
+		case CS7: urb_value |= 7; dbg("Setting CS7"); break;
+		case CS8: urb_value |= 8; dbg("Setting CS8"); break;
+		default:
+			dev_err(&port->dev, "CSIZE was set but not CS7-CS8\n");
+		}
 	}
 
 	/* This is needed by the break command since it uses the same command
-- 
1.7.7

